modules/carto/src/basemap.ts

```diff
@@ -1,4 +1,97 @@
-const baseUrl = 'https://basemaps.cartocdn.com/gl/{basemap}-gl-style/style.json';
+export const cartoBasemapsBaseUrl = 'https://basemaps.cartocdn.com/gl';
+
+const baseUrl = `${cartoBasemapsBaseUrl}/{basemap}-gl-style/style.json`;
+
+export const cartoStyleLayerGroups = [
+  {
+    slug: 'label',
+    filter: ({id}: {id: string}) =>
+      Boolean(
+        id.match(/(?=(label|_label|place-|place_|poi-|poi_|watername_|roadname_|housenumber))/)
+      ),
+    defaultVisibility: true
+  },
+  {
+    slug: 'road',
+    filter: ({id}: {id: string}) =>
+      Boolean(id.match(/(?=(road|railway|tunnel|street|bridge))(?!.*label)/)),
+    defaultVisibility: true
+  },
+  {
+    slug: 'border',
+    filter: ({id}: {id: string}) => Boolean(id.match(/border|boundaries|boundary_/)),
+    defaultVisibility: false
+  },
+  {
+    slug: 'building',
+    filter: ({id}: {id: string}) => Boolean(id.match(/building/)),
+    defaultVisibility: true
+  },
+  {
+    slug: 'water',
+    filter: ({id}: {id: string}) => Boolean(id.match(/(?=(water|stream|ferry))/)),
+    defaultVisibility: true
+  },
+  {
+    slug: 'land',
+    filter: ({id}: {id: string}) =>
+      Boolean(id.match(/(?=(parks|landcover|industrial|sand|hillshade|park_))/)),
+    defaultVisibility: true
+  }
+];
+
+export function applyCartoLayerGroupFilters(style, visibleLayerGroups: Record<string, boolean>) {
+  if (!Array.isArray(style?.layers)) {
+    return style;
+  }
+
+  const visibleFilters = cartoStyleLayerGroups
+    .filter(lg => visibleLayerGroups[lg.slug])
+    .map(lg => lg.filter);
+
+  const filteredLayers = style.layers.filter(layer => visibleFilters.some(match => match(layer)));
+
+  return {
+    ...style,
+    layers: filteredLayers
+  };
+}
+
+function someLayersFilteredOut(visibleLayerGroups?: Record<string, boolean>) {
+  return (
+    visibleLayerGroups &&
+    Object.values(visibleLayerGroups).some(filteredOut => filteredOut === false)
```

Comment:
I was a little confused to read a function named `someLayersFilteredOut` returning true if some layers have `filteredOut === false`, perhaps `visible` is clearer?

```suggestion
    Object.values(visibleLayerGroups).some(visible => visible === false)
```

Alternatively, maybe:

```javascript
if (visibleLayerGroups && !Object.values(visibleLayerGroups).every(Boolean)) {
  // ...
}

```